### PR TITLE
Wire up the project config into the lint command

### LINKED ISSUE #56: `lint` with no args doesn't display a helpful message
This doesn't make it clear that a file name is needed:

```
  polylint checks Polymer apps for problematic code patterns

polylint

Options

  -h, --help              Print usage.                                                                  
  -b, --bowerdir string   Bower components directory. Defaults to 'bower_components'                    
  -v, --verbose           Writes verbose logging.                                                       
  -g, --debug             Writes debugging trace.                                                       
  -p, --policy string     Your jsconf.json policy file.                                                 
  -r, --root string       Root directory against which URLs in inputs are resolved. If not specified,   
                          then the current working directory is used.                                   
  -i, --input string[]    Polymer source files.                                                         
  --config-file string    If inputs are specified, look for `config-field` in this JSON file.           
  --config-field string   If config-file is used for inputs, this field determines which file(s) are    
                          linted.                                                                       
  --stdin                 If true, the file from `input` will be replaced by the contents of stdin. If  
                          true, only one value will be accepted for input.                              
  --no-recursion          Only report errors on specified input files, not from their dependencies.     
```

### PR DESCRIPTION
**Pull request by justinfagnani**
Fixes #56 and #83

### CODE REVIEW

File: src/commands/lint.ts [hunk 1]
```diff
@@ -52,7 +54,25 @@ export class LintCommand implements Command {
   ];
 
   run(options, config): Promise<any> {
-    return polylint.runWithOptions(options)
-        .then(() => null);
+    if (config.inputs.length === 0) {
+      let argsCli = commandLineArgs(this.args);
+      // TODO: use logging
+      console.log('No inputs specified. Please use the --entrypoint, --shell ' +
```
Comment (FredKSchott) [inline src/commands/lint.ts]: use our new logging system! `console.log` -> `logger.info`

File: src/project-config.ts [hunk 2]
```diff
@@ -38,10 +39,14 @@ export class ProjectConfig {
   }
 
   constructor(configFile?: string, options?: ProjectConfigOptions) {
-    this.init(configFile, options);
+    this._init(configFile, options);
   }
 
-  init(configFile?: string, options?: ProjectConfigOptions) {
+  isInput(path: string): boolean {
```
Comment (azakus) [inline src/project-config.ts]: Unused?

File: src/polymer-cli.ts [hunk 3]
```diff
@@ -22,6 +22,16 @@ import {ProjectConfig, ProjectConfigOptions} from './project-config';
 import {Environment} from './environment/environment'
 import {buildEnvironment} from './environments/environments'
 
+process.on('uncaughtException', (error) => {
+  console.log(`Uncaught exception: ${error}`);
```
Comment (FredKSchott) [inline src/polymer-cli.ts]: `console.log` -> `logger.info`
`console.error` -> `logger.error`

File: src/commands/lint.ts [hunk 4]
```diff
@@ -52,7 +54,25 @@ export class LintCommand implements Command {
...
+          'or --fragment flags');
+      console.log(argsCli.getUsage({
+        title: `polymer ${this.name}`,
+        description: this.description,
+      }));
+    }
+    return polylint.runWithOptions({
```
Comment (FredKSchott) [inline src/commands/lint.ts]: 👍 👍 👍

File: src/commands/lint.ts (hunk 1 above)
Comment (justinfagnani) [inline src/commands/lint.ts]: done

File: src/project-config.ts (hunk 2 above)
Comment (justinfagnani) [inline src/project-config.ts]: done

File: src/commands/lint.ts [hunk 5]
```diff
@@ -52,7 +54,25 @@ export class LintCommand implements Command {
...
-        .then(() => null);
+    if (config.inputs.length === 0) {
+      let argsCli = commandLineArgs(this.args);
+      // TODO: use logging
+      console.log('No inputs specified. Please use the --entrypoint, --shell ' +
+          'or --fragment flags');
+      console.log(argsCli.getUsage({
+        title: `polymer ${this.name}`,
+        description: this.description,
+      }));
```
Comment (FredKSchott) [inline src/commands/lint.ts]: missing return?

File: src/polymer-cli.ts (hunk 3 above)
Comment (justinfagnani) [inline src/polymer-cli.ts]: done

File: src/commands/lint.ts (hunk 5 above)
Comment (justinfagnani) [inline src/commands/lint.ts]: done

**Comment by FredKSchott** [comment]
LGTM besides that missing return comment!

**Comment by justinfagnani** [comment]
Fixed the things

**Comment by FredKSchott** [comment]
LGTM

**Pull request merged**